backend-shared/config/db/util/utils.go

```diff
@@ -109,22 +109,22 @@ func GetOrCreateManagedEnvironmentByNamespaceUID(ctx context.Context, workspaceN
 	}
 
 	if err := dbq.CreateKubernetesResourceToDBResourceMapping(ctx, dbResourceMapping); err != nil {
-		return nil, fmt.Errorf("unable to create KubernetesResourceToDBResourceMapping: %v", err)
+		return nil, fmt.Errorf("unable to create KubernetesResourceToDBResourceMapping: %v", err), false
 	}
 
-	return &managedEnvironment, nil
+	return &managedEnvironment, nil, true
 }
 
 // GetOrCreateGitopsEngineInstanceByInstanceNamespaceUID gets (or creates it if it doesn't exist) a GitOpsEngineInstance database entry that
 // corresponds to an GitOps engine instance running on the cluster.
 func GetOrCreateGitopsEngineInstanceByInstanceNamespaceUID(ctx context.Context,
 	gitopsEngineNamespace v1.Namespace, kubesystemNamespaceUID string,
-	dbq db.DatabaseQueries, log logr.Logger) (*db.GitopsEngineInstance, *db.GitopsEngineCluster, error) {
+	dbq db.DatabaseQueries, log logr.Logger) (*db.GitopsEngineInstance, bool, *db.GitopsEngineCluster, error) {
```

Comment:
As above, we can add a quick comment to the function comment, indicating what the bool is for.